pkg/apis/maistra/conversion/discovery_selectors.go

```diff
@@ -0,0 +1,51 @@
+package conversion
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "github.com/maistra/istio-operator/pkg/apis/maistra/v1"
+	v2 "github.com/maistra/istio-operator/pkg/apis/maistra/v2"
+)
+
+func populateDiscoverySelectorsValues(in *v2.ControlPlaneSpec, out map[string]interface{}) error {
+	if in.MeshConfig == nil || in.MeshConfig.DiscoverySelectors == nil {
+		return nil
+	}
+
+	untypedSlice := make([]interface{}, len(in.MeshConfig.DiscoverySelectors))
+	for index, value := range in.MeshConfig.DiscoverySelectors {
+		untypedSlice[index] = value
+	}
+	if discoverySelectors, err := sliceToValues(untypedSlice); err == nil {
+		if err := setHelmValue(out, "meshConfig.discoverySelectors", discoverySelectors); err != nil {
+			return err
+		}
+	} else {
+		return err
+	}
+
+	return nil
+}
+
+func populateDiscoverySelectorsConfig(in *v1.HelmValues, out *v2.ControlPlaneSpec) error {
+	setDiscoverySelectors := false
+
+	discoverySelectors := []*metav1.LabelSelector{}
+	if ds, ok, err := in.GetFieldNoCopy("meshConfig.discoverySelectors"); ok {
+		if err := fromValues(ds, &discoverySelectors); err != nil {
+			return err
+		}
+		setDiscoverySelectors = true
+		in.RemoveField("meshConfig.discoverySelectors")
+	} else if err != nil {
+		return err
+	}
+
+	if setDiscoverySelectors {
+		if out.MeshConfig == nil {
+			out.MeshConfig = &v2.MeshConfig{}
+		}
+		out.MeshConfig.DiscoverySelectors = discoverySelectors
+	}
```

Comment:
You can remove this boolean variable and simplify function as below:
```suggestion
        var discoverySelectors []*metav1.LabelSelector
	if ds, ok, err := in.GetAndRemoveSlice("meshConfig.discoverySelectors"); ok {
		if err := fromValues(ds, &discoverySelectors); err != nil {
			return err
		}
		if out.MeshConfig == nil {
			out.MeshConfig = &v2.MeshConfig{}
		}
		out.MeshConfig.DiscoverySelectors = discoverySelectors
	} else if err != nil {
		return err
	}
```